server/src/main/kotlin/delta/codecharacter/server/match/MatchService.kt

```diff
@@ -312,17 +359,37 @@ class MatchService(
                 matchRepository.save(finishedMatch)
             }
         } else if (dailyChallengeMatchRepository.findById(matchId).isPresent) {
-            println(updatedGame.destruction)
-            println(updatedGame.matchId)
-      /*
-       * Get the DCMatch with help of the matchId
-       * pass the game parameters into the daily-challenge-verdict-algorithm
-       * Game-Parameters :- GameStatus, destruction,coinsUsed
-       * Store the verdict back in verdict in daily-challenge entity
-       * websocket for sending game status /updates/{userId} -> userId from dcMatchEntity
-       * store the finsished DC Match
-       * If verdict-> success make isDailyChallengeComplete-true
-       */
+            val match = dailyChallengeMatchRepository.findById(matchId).get()
+            simpMessagingTemplate.convertAndSend(
+                "/updates/${match.user.userId}",
+                mapper.writeValueAsString(
+                    GameDto(
+                        id = updatedGame.id,
+                        destruction = BigDecimal(updatedGame.destruction),
+                        coinsUsed = updatedGame.coinsUsed,
+                        status = GameStatusDto.valueOf(updatedGame.status.name),
+                    )
+                )
+            )
+            if (updatedGame.status != GameStatusEnum.EXECUTING) {
+                val updatedMatch =
+                    match.copy(
+                        verdict =
+                        dailyChallengeService.completeDailyChallenge(updatedGame, match.user.userId)
+                    )
+                notificationService.sendNotification(
+                    match.user.userId,
+                    title = "Daily Challenge Results",
+                    content =
+                    when (updatedMatch.verdict) {
+                        DailyChallengeMatchVerdictEnum.SUCCESS -> "Successfully completed challenge"
+                        else -> {
+                            "Failed to complete challenge"
+                        }
+                    }
```

Comment:
Add `DailyChallengeMatchVerdictEnum.FAILURE` separately and in `else` mention `Some error occured. Try again!`.